Revamp: Grid Filtering (#2299)
telerik/blazor-docs: components/grid/filter/overview.md

position: 0
---

# Filtering Overview

This article explains the available filtering modes in the Telerik Grid for Blazor.

## FilterRow

The `FilterRow` filtering mode renders a row below the column headers with a UI for filling in the filter criteria. Read more about enabling and fine-tuning the filtering row in the [FilterRow documentation article...]({%slug grid-filter-row%})

## FilterMenu

The FilterMenu filter mode renders a button in the column header. Clicking the button opens a popup with filtering options, allowing you to apply two filter criteria, choose a filter operator, and use buttons to apply or clear the filter. Read more about enabling and fine-tuning the filtering menu in the [FilterMenu documentation article...]({%slug grid-filter-menu%})

> You can prevent the user from filtering a certain field by setting `Filterable="false"` on its column.

## More Filtering Options

In addition to the two main filtering modes, the grid offers two more features that can enhance the user experience when looking for data.

### ToolBar SearchBox

> The ToolBar of the Telerik Grid for Blazor includes a [SearchBox]({%slug grid-searchbox%}), that lets users amend filters and search across multiple fields simultaneously.

Reviewer: ```suggestion
The ToolBar of the Telerik Grid for Blazor includes a [SearchBox]({%slug grid-searchbox%}) that lets users amend filters and search across multiple fields simultaneously.
```

### CheckBoxList

The filter menu can display a [list of checkboxes]({%slug grid-checklist-filter%}) with distinct values from the data, making filtering similar to Excel.

## Filter Descriptors

> The Grid filter state is stored in [CompositeFilterDescriptors](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor). Use the following information if you want to [get or change the Grid filters programmatically]({%slug grid-state%}).

Reviewer: ```suggestion
The Grid filter state is stored in [`CompositeFilterDescriptors`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor). Use the following information if you want to [get or change the Grid filters programmatically]({%slug grid-state%}).
```

> Each CompositeFilterDescriptor includes a collection of FilterDescriptors, where all descriptors in the collection are applied with an AND or OR LogicalOperator.

Reviewer: Don't invent new words when referring to well-known concepts&mdash;follow the common writing standards.
```suggestion
Each `CompositeFilterDescriptor` includes a collection of filter descriptors, where all descriptors in the collection are applied with an AND or OR logical operator.
```

> * [Filter Row]({%slug grid-filter-row%}) - Each `CompositeFilterDescriptor` targets a specific field. By default, one filter can be applied to a field using the Filter Row operator, with the filter value stored in the first FilterDescriptor instance for that field.

Reviewer: Use an em dash. The same comment applies to the next two lines.
```suggestion
* [Filter Row]({%slug grid-filter-row%})&mdash;Each `CompositeFilterDescriptor` targets a specific field. By default, one filter can be applied to a field using the Filter Row operator, with the filter value stored in the first filter descriptor instance for that field.
```

> * [Filter Menu]({%slug grid-filter-menu%}) - Each `CompositeFilterDescriptor` targets a specific field. The filter values from separate filter operators in the menu are stored in different FilterDescriptor instances within the CompositeFilterDescriptor for that field.

Reviewer: ```suggestion
* [Filter Menu]({%slug grid-filter-menu%})&mdash;Each `CompositeFilterDescriptor` targets a specific field. The filter values from separate filter operators in the menu are stored in different filter descriptor instances within the `CompositeFilterDescriptor` for that field.
```

> * [SearchBox]({%slug grid-searchbox%}) - A `CompositeFilterDescriptor` is created in the state when the user types in the Searchbox. By default, it targets all string fields, adding a dedicated FilterDescriptor instance for each string field. Each FilterDescriptor contains the filter value typed in the SearchBox.

Reviewer: ```suggestion
* [SearchBox]({%slug grid-searchbox%})&mdash;A `CompositeFilterDescriptor` is created in the state when the user types in the search box. By default, it targets all string fields, adding a dedicated filter descriptor instance for each string field. Each filter descriptor contains the filter value typed in the search box.
```

## Notes

> The grid uses `Activator.CreateInstance<TItem>();` to obtain the item's type it is bound to, enabling it to generate accurate filters and filter operators. To facilitate this, ensure that the Model has a parameterless constructor defined. If your model has no parameterless constructor use the [OnModelInit event]({%slug grid-events%}#onmodelinit).

Reviewer: Capital "G"
```suggestion
The Grid uses `Activator.CreateInstance<TItem>();` to obtain the item's type it is bound to, enabling it to generate accurate filters and filter operators. To facilitate this, ensure you define a parameterless constructor for the model. If your model has no parameterless constructor use the [`OnModelInit` event]({%slug grid-events%}#onmodelinit).
```
